### ISSUE TITLE
[BUG] `avoid-unused-parameters` conflicts with `avoid_renaming_method_parameters`

### ISSUE THREAD

**Issue by grafovdenis** [dart-code-checker/dart-code-metrics#372]
* **Dart code metrics version:** 3.3.3
* **Dart sdk version:** 2.13

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
analyzer:
  plugins:
    - dart_code_metrics

dart_code_metrics:
  rules:
    - avoid-unused-parameters

linter:
  rules:
    - avoid_renaming_method_parameters: true
```

</details>

**What did you do? Please include the source code example causing the issue.**

When I write any widget using Flutter, I recieve `avoid-unused-parameters` lint in widget's `build` method on `BuildContext context` param. When I fix this waring, I recieve `avoid_renaming_method_parameters`on `_` or `BuildContext  _`.

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(_) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}
```

I suggest you to add kinda `ignore` param to `avoid-unused-parameters` to ignore this rule with some params by name or by type.

**Comment by roman-petrov** [comment]
I also faced the same problem in our project. I suggest to add a parameter to allow to ignore overridden methods since I think it is nothing bad in not using parameters in overridden methods

**Comment by incendial** [comment]
I think we should just disable this check for overridden methods, because all other solutions looks bad to me.

**Comment by incendial** [comment]
Or, yeah, maybe an `ignore` parameter that is enabled by default.

**Comment by incendial** [comment]
Published 3.3.5 with this incompatibility fix.
